crates/compiler/solve/src/solve.rs

```diff
@@ -1771,6 +1771,57 @@ fn solve(
 
                 state
             }
+            IngestedFile(type_index, _, bytes) => {
+                let actual = either_type_index_to_var(
+                    subs,
+                    rank,
+                    pools,
+                    problems,
+                    abilities_store,
+                    obligation_cache,
+                    &mut can_types,
+                    aliases,
+                    *type_index,
+                );
+
+                if let Success { .. } = unify(
+                    // TODO: if we don't clone and a later branch matches, we will get a failure in alias analysis.
+                    // That said, I assume cloning is expensive and should be avoided.
+                    &mut UEnv::new(&mut subs.clone()),
+                    actual,
+                    Variable::LIST_U8,
+                    Mode::EQ,
+                    Polarity::OF_VALUE,
+                ) {
+                    // List U8 always valid.
+                    state
+                } else if let Success { .. } = unify(
```

Comment:
You can use `subs.snapshot` here and rollback if it fails. It's used all over the place elsewhere.

The variables in `Success` cannot be ignored. If unification succeeds, they need to be processed - take a look at how the `Eq` constraint is implemented for what needs to happen with those values.

Reply:
I know I wrote a reply to this already, but I guess I forgot to send it or it got lost somehow. Anyway.

I don't think we actually need to truly unify here. I have changed the code so that no matter what, the subs are rolled back. The type has an equal constraint that runs before the ingested file constraint. That `Eq` constraint should fully define the ingested file to 1 specific type. Then when we get this code, we aren't really trying to unify the type. We are just trying to check what the type is. If I could convert the type to a layout here and just switch on the layout, I would.